siliconcompiler/core.py

```diff
@@ -4935,32 +4918,63 @@ def _is_builtin(self, tool, task):
         return task in self.builtin
 
     #######################################
-    def _get_steps_by_task(self, flow=None):
+    def _get_flowgraph_entry_nodes(self, flow=None):
+        '''
+        Collect all step/indecies that represent the entry
+        nodes for the flowgraph
+        '''
+        if not flow:
+            flow = self.get('option', 'flow')
+
+        nodes = []
+        for step in self.getkeys('flowgraph', flow):
+            for index in self.getkeys('flowgraph', flow, step):
+                if not self.get('flowgraph', flow, step, index, 'input'):
+                    nodes.append((step, index))
+        return nodes
+
+    #######################################
+    def _get_flowgraph_exit_nodes(self, flow=None):
```

Comment:
I think this function is basically the same as `Chip._find_leaves()`, could we get rid of that one/merge them?

Reply:
merged into one function